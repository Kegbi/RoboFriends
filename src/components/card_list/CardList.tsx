import React from "react";
import Card from "./card/Card";
import { IRobotsState } from "../../types/types";

function CardList({ robots }: IRobotsState) {
	return (
		<div>
			{robots.map((user, i) => {
				return (
					<Card
						key={robots[i].id}
						id={robots[i].id}
						name={robots[i].name}
						email={robots[i].email}
					/>
				);
			})}
		</div>
	);
}

export default CardList;
