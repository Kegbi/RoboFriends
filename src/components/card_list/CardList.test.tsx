import React from "react";
import CardList from "./CardList";
import renderer from 'react-test-renderer';

it('Expect to render CardList component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John Snow',
      username: 'johnsnow',
      email: 'john@gmail.com'
    }
  ]
  const tree = renderer.create(<CardList robots={mockRobots}/>).toJSON()
  expect(tree).toMatchSnapshot();
})