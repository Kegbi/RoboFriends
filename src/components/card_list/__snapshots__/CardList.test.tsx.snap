// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render CardList component 1`] = `
<div>
  <div
    className="card"
  >
    <img
      alt="kitty"
      src="https://robohash.org/1?set=set3"
    />
    <h2
      className="card__header"
    >
      John Snow
    </h2>
    <p
      className="card_mail"
    >
      john@gmail.com
    </p>
  </div>
</div>
`;
